files/usr/etc/profile.d/ssh-locale-check.sh

```diff
@@ -0,0 +1,14 @@
+#
+# locale-check.sh: Verify that the locale SSH did set is valid,
+#                  else reset to the system default.
+#
+
+# Only check locale if it did got set by SSH
+test -z "$SSH_SENDS_LOCALE" && return
+
+_SYSTEM_DEFAULT_LANG=C.UTF-8
+if [ -s /etc/locale.conf ]; then
+    eval "$(sed -rn -e 's/^(LANG)=/_SYSTEM_DEFAULT_\1=/p' < /etc/locale.conf)"
```

Comment:
since we already do have a C program at hand this extra step through sed could be avoided if the C program itself reads the file

Reply:
In that case you would use all flexibility from it and you cannot specify any other locale for that case. Only because the shell hook is for ssh, you could also use the program for other use cases or specify other defaults for group of users.

Reply:
pretty easy, if argc < 2 read /etc/locale.conf. That's basically what your shell wrapper does, compute argv[1]